src/vs/base/browser/ui/list/listWidget.ts

```diff
@@ -529,14 +529,21 @@ class TypeNavigationController<T> implements IDisposable {
 
 			if (this.list.options.typeNavigationEnabled) {
```

Comment:
I'd like to simplify the if/else checks, but the reason for the initial check is for when `typeNavigationEnabled` is true. The `else` checks all other cases (so that your initial code is untouched). The check for when `labelStr` is undefined is under the check for if `typeNavigationEnabled` is true because in the action widget, there are instances where `labelStr` could be undefined and we don't want it to do anything.